src/test/ui/consts/const-eval/const-pointer-values-in-various-types.rs

```diff
@@ -37,7 +37,7 @@ fn main() {
     //~^ ERROR this constant likely exhibits undefined behavior
 
     const I32_REF_U128_UNION: u128 = unsafe { Nonsense { int_32_ref: &3 }.uint_128 };
-    //~^ ERROR this constant cannot be used
+    //~^ ERROR this constant likely exhibits undefined behavior
```

Comment:
Why did this go from "evaluation failed" to "validation failed"?

Reply:
because evaluation used to fail in order to produce `ConstValue::Scalar(Scalar::Bits { size: 128, value: UNDEF })` which is clearly not possible.

Now this is simply `ByRef` and thus can contain undef, which will later be caught by validation

Reply:
Ah I see, this got normalized so there was an undef error earlier on. Makes sense, and nice improvement!